sdk/src/Core/RegionEndpoint/RegionEndpointProviderV3.cs

```diff
@@ -92,7 +94,7 @@ public RegionEndpoint.Endpoint GetEndpointForService(string serviceName, HashSet
         {
             RegionEndpoint.Endpoint endpointObject = null;
 
-            lock (_serviceMap)
+            lock (_serviceMapLock)
```

Comment:
The reason for this page is the [lock documentation](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock): _

> Avoid using the same lock object instance for different shared resources, as it might result in deadlock or lock contention.

_